tools/util.wdl

```diff
@@ -353,7 +353,7 @@ task unpack_tarball {
 
         mkdir unpacked_tarball
         tar -C unpacked_tarball -xzf ~{tarball} --no-same-owner
-        find unpacked_tarball/ -type f > file_list.txt
+        find unpacked_tarball/ -type f | LC_ALL=C sort > file_list.txt
```

Comment:
When the "find order" bug was first identified, we discussed this exact fix. And we opted not to implement it. Not 100% certain this was the original rationale, but I think we decided it "wasn't the task's job" to have sorted output.  Or rather, having a `sort` statement would raise questions about "why". I don't think I still agree with that though.

I'm fine with this, but just documenting the historical rationale. Please delete the comment about this in the `yaml` file though.

Reply:
I don't fully recall that conversion, but I will note that one of the `multiqc` tasks explicitly sets `LC_ALL`. So this wouldn't be the only one. 

That said, I dug into this and it seems that `find` has no inherent sorting. It is fully up to the order that the filesystem returns items. For all practical purposes (other than this test), that should be fine. I would hope that no downstream task would `untar` many files and then try to address them by index. 

For example, here is the output of untarring on the cluster.

```
$ tar -C unpacked_tarball -vxzf test.tar.gz --no-same-owner
test/
test/b
test/a

$ find unpacked_tarball/ -type f
unpacked_tarball/test/b
unpacked_tarball/test/a
```

We can see that `tar` processes `b` and then `a` and the `find` command respects that ordering. 

For another example, I ran this on my Mac in a Ubuntu Docker container.

```
# tar -C unpacked_tarball -vxzf /data/tests/tools/input/test.tar.gz --no-same-owner
test/
test/b
test/a

# find unpacked_tarball/ -type f
unpacked_tarball/test/a
unpacked_tarball/test/b
```

Again `tar` processes `b` and then `a`, but the `find` command reverses that ordering.

Looking at `stat` output on the cluster, the inode and timestamps (except `modify`) all make sense for `b` preceding `a`.

```
$ stat unpacked_tarball/test/a
  File: unpacked_tarball/test/a
  Size: 4               Blocks: 1          IO Block: 4194304 regular file
Device: 34h/52d Inode: 14406740294  Links: 1
Access: (0770/-rwxrwx---)  Uid: (1018972/athrashe)   Gid: (76937/Domain Users)
Access: 2024-01-31 15:37:04.258968000 -0600
Modify: 2023-09-20 11:15:13.000000000 -0500
Change: 2024-01-31 15:37:04.259195840 -0600
 Birth: -

$ stat unpacked_tarball/test/b
  File: unpacked_tarball/test/b
  Size: 4               Blocks: 1          IO Block: 4194304 regular file
Device: 34h/52d Inode: 14406740293  Links: 1
Access: (0770/-rwxrwx---)  Uid: (1018972/athrashe)   Gid: (76937/Domain Users)
Access: 2024-01-31 15:37:04.258462000 -0600
Modify: 2023-09-20 11:15:24.000000000 -0500
Change: 2024-01-31 15:37:04.258663107 -0600
 Birth: -
```

On my mac, the stat is similar (other than timezone). 

```
# stat unpacked_tarball/test/a
  File: unpacked_tarball/test/a
  Size: 4         	Blocks: 8          IO Block: 4096   regular file
Device: 88h/136d	Inode: 3676512     Links: 1
Access: (0750/-rwxr-x---)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2024-01-31 21:38:23.650214005 +0000
Modify: 2023-09-20 16:15:13.000000000 +0000
Change: 2024-01-31 21:38:23.651214005 +0000
 Birth: 2024-01-31 21:38:23.650214005 +0000

# stat unpacked_tarball/test/b
  File: unpacked_tarball/test/b
  Size: 4         	Blocks: 8          IO Block: 4096   regular file
Device: 88h/136d	Inode: 3676511     Links: 1
Access: (0750/-rwxr-x---)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2024-01-31 21:38:23.649214005 +0000
Modify: 2023-09-20 16:15:24.000000000 +0000
Change: 2024-01-31 21:38:23.650214005 +0000
 Birth: 2024-01-31 21:38:23.649214005 +0000
```

Yet the output order of `find` is reversed. So the difference must be something in the implementation of GPFS vs APFS. I've tried options such as `tar`'s `--preserve-order`, but it doesn't seem to be `tar` output ordering, it is the underlying file system ordering.

Reply:
Looking at this again, I'm less convinced that omitting a `sort` is proper. We're not returning the structure of the `tar` file. We're explicitly deconstructing it and returning only a listing of files that were contained. Without the directory structure, does that make sense to return an unsorted list?

Reply:
Ahh I remembered more of the original conversation thanks to this sentence:

> I would hope that no downstream task would untar many files and then try to address them by index.

The rationale was "if we guarantee an output order, regardless of what that order is, we might encourage users to try indexing into the produced array looking for file X. To prevent that, let's leave the order up to the filesystem which is inherently unreliable, preventing that use case".

I don''t agree with that anymore. My current thinking: if we _don't_ pass it through `sort`, we need to write a custom test. If we _do_ pass it through sort, our current test works. I do not want to write a custom test. Let's just pass it through `sort` and call it a day.

But I'd make one last change request. Can we put a comment which says `# pipe through sort because otherwise order is random (dependent on filesystem)` or something similar?

Reply:
> Please delete the comment about this in the yaml file though.

https://github.com/stjudecloud/workflows/blob/main/tests/tools/test_util.yaml#L85

^that comment should be deleted now that we've settled this behavior.

After that I'll approve the changes